modules/custom/az_migration/src/Plugin/migrate/process/DateTimeToSmartDate.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Process Plugin to handle migrating Drupal datetime field to smart_date field.
+ *
+ * Expects a multidimentional array typical of a Drupal datetime field.
+ *
+ * Available configuration keys
+ * - source_start: (required) Field to copy start time from.
+ * - source_end: (optional) Field to copy end time from, if not set the
+ *   source_start field value will be used.
+ * - default_duration: (int, optional) If no end date, assumed duration.
+ *
+ * Consider a datetime field migration
+ * @code
+ *   process:
+ *     field_az_event_date:
+ *       - plugin: single_value
+ *         source: field_uaqs_event_date
+ *       - plugin: az_drupal_date_to_smart_date
+ *         source_start: value
+ *         source_end: value2
+ *         timezone: 'America/Phoenix'
+ * @endcode
+ *
+ * @MigrateProcessPlugin(
+ *   id = "az_datetime_to_smart_date"
+ * )
+ */
+class DateTimeToSmartDate extends ProcessPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    if (empty($this->configuration['source_start'])) {
+      throw new InvalidPluginDefinitionException(
+        $this->getPluginId(),
+        "Configuration option 'source_start' is required."
+      );
+    }
+    if (isset($this->configuration['default_duration']) && !is_numeric($this->configuration['default_duration'])) {
+      throw new InvalidPluginDefinitionException(
+        $this->getPluginId(),
+        "Configuration option 'default_duration' should be a numeric represention of a length of time in minutes,
+        for example 60 is one minute."
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+
+    return $instance;
+  }
```

Comment:
Since this doesn't appear to be adding any additional dependencies, I don't think it's actually necessary to implement tis `create()` method...
```suggestion
```

Reply:
```
     Child process error (exit code 255): PHP Fatal error:  Class               
11     Drupal\az_migration\Plugin\migrate\process\DateTimeToSmartDate contains 1  
12     abstract method and must therefore be declared abstract or implement the   
13     remaining methods                                                          
14     (Drupal\Core\Plugin\ContainerFactoryPluginInterface::create) in            
15     /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/  
16     custom/az_migration/src/Plugin/migrate/process/DateTimeToSmartDate.php on  
17     line 38                                                                    
18     Fatal error: Class                                                         
19     Drupal\az_migration\Plugin\migrate\process\DateTimeToSmartDate contains 1  
20     abstract method and must therefore be declared abstract or implement the   
21     remaining methods                                                          
22     (Drupal\Core\Plugin\ContainerFactoryPluginInterface::create) in            
23     /src/az-quickstart-scaffolding/web/profiles/custom/az_quickstart/modules/  
24     custom/az_migration/src/Plugin/migrate/process/DateTimeToSmartDate.php on  
25     line 38                                                                    
```